cmd/export_orderbooks.go

```diff
@@ -89,42 +88,76 @@ var exportOrderbooksCmd = &cobra.Command{
 				batchStart := startNum + batchNum*batchSize
 				batchEnd := batchStart + batchSize - 1
 				parser := input.ReceiveParsedOrderbooks(orderbookChannel, cmdLogger)
-				exportOrderbook(batchStart, batchEnd, outputFolder, parser, gcpCredentials, gcsBucket, runId)
+				exportOrderbook(batchStart, batchEnd, outputFolder, parser, gcpCredentials, gcsBucket, extra)
 				batchNum++
 			}
 		}
 	},
 }
 
 // writeSlice writes the slice either to a file.
-func writeSlice(file *os.File, slice [][]byte) {
-	for _, v := range slice {
-		file.WriteString(string(v) + "\n")
+func writeSlice(file *os.File, slice [][]byte, extra map[string]string) error {
+
+	for _, data := range slice {
+		bytesToWrite := data
+		if len(extra) > 0 {
+			i := map[string]interface{}{}
+			err := json.Unmarshal(data, &i)
+			if err != nil {
+				return err
+			}
+			for k, v := range extra {
+				i[k] = v
+			}
+			bytesToWrite, err = json.Marshal(i)
+			if err != nil {
+				return err
+			}
```

Comment:
❓ if the `export_orderbooks` is processing multiple batches, does this mean that there could be multiple lines of run metadata (`extra`) written to the same file? Ie would there be a metadata record _per_ batch in the file?

Reply:
The extra fields are appended to every single row in the output, since they're associated with every run in BigQuery. It is not per batch, but per row.

Reply:
Tracking now. I has assumed this was the `{transforms: 20, failed_transforms:0, etc}` record. Makes sense